src/main.cpp

```diff
@@ -223,40 +228,66 @@ void publishMsgs(um7::Registers& r, ros::NodeHandle* imu_nh, sensor_msgs::Imu& i
 
   if (imu_pub.getNumSubscribers() > 0)
   {
-    // body-fixed frame NED to ENU: (x y z)->(x -y -z) or (w x y z)->(x -y -z w)
-    // world frame      NED to ENU: (x y z)->(y  x -z) or (w x y z)->(y  x -z w)
-    if (tf_ned_to_enu)
-    {
-      // world frame
-      imu_msg.orientation.w =  r.quat.get_scaled(2);
-      imu_msg.orientation.x =  r.quat.get_scaled(1);
-      imu_msg.orientation.y = -r.quat.get_scaled(3);
-      imu_msg.orientation.z =  r.quat.get_scaled(0);
-
-      // body-fixed frame
-      imu_msg.angular_velocity.x =  r.gyro.get_scaled(0);
-      imu_msg.angular_velocity.y = -r.gyro.get_scaled(1);
-      imu_msg.angular_velocity.z = -r.gyro.get_scaled(2);
-
-      // body-fixed frame
-      imu_msg.linear_acceleration.x =  r.accel.get_scaled(0);
-      imu_msg.linear_acceleration.y = -r.accel.get_scaled(1);
-      imu_msg.linear_acceleration.z = -r.accel.get_scaled(2);
-    }
-    else
+    switch (axes)
     {
-      imu_msg.orientation.w = r.quat.get_scaled(0);
-      imu_msg.orientation.x = r.quat.get_scaled(1);
-      imu_msg.orientation.y = r.quat.get_scaled(2);
-      imu_msg.orientation.z = r.quat.get_scaled(3);
-
-      imu_msg.angular_velocity.x = r.gyro.get_scaled(0);
-      imu_msg.angular_velocity.y = r.gyro.get_scaled(1);
-      imu_msg.angular_velocity.z = r.gyro.get_scaled(2);
-
-      imu_msg.linear_acceleration.x = r.accel.get_scaled(0);
-      imu_msg.linear_acceleration.y = r.accel.get_scaled(1);
-      imu_msg.linear_acceleration.z = r.accel.get_scaled(2);
+        case ENU:
```

Comment:
@bhomberg it looks like the indentation was 2 spaces and you've changed it to 4 in some places. Could you change it back? That should make it easier for me to see what has actually changed.

Reply:
@jonbinney fixed. Ready for a review.